server/netty-server/sync/src/main/scala/sttp/tapir/server/netty/sync/internal/reactivestreams/OxProcessor.scala

```diff
@@ -63,11 +71,15 @@ private[sync] class OxProcessor[A, B](
       val channelSubscription = new ChannelSubscription(wrappedSubscriber, outgoingResponses)
       subscriber.onSubscribe(channelSubscription)
       channelSubscription.runBlocking() // run the main loop which reads from the channel if there's demand
-    } { error =>
+    }(pipelineFork) { error =>
       wrappedSubscriber.onError(error)
       onError(error)
     }
 
+  private def cancelPipelineFork(): Unit =
+    try { pipelineFork.get(pipelineCancelationTimeoutMs, TimeUnit.MILLISECONDS).cancelNow() }
```

Comment:
The only problem here might be that the fork might never actually cancel. Not sure if we want to handle this ... one way would be to submit to OxDispatcher a task which would race: `sleep(timeout); log.warn("Fork doesn't want to die")` and `pipelineFork.cancel()`. This could also have some performance implications, though.

Reply:
I tested this `race` idea with performance tests and I didn't notice latency degradation. However, `race` returns a `Left(InterruptedException)`. I guess I should ignore it, because it means that `sleep` got interrupted? Or it rather shouldn't happen?

Reply:
I guess that's the result of `CancellableFork.cancel()`? Which returns an interrupted exception, as expected.